### ISSUE TITLE
Crash when device disk is full

### ISSUE THREAD

**Issue by laalto** [urbanairship/android-library#111]
# Preliminary Info

### What Urban Airship dependencies are you using?

com.urbanairship.android:urbanairship-fcm:9.7.0
com.urbanairship.android:urbanairship-core:9.7.0

### What are the versions of any relevant development tools you are using?

N/A

# Report

### What unexpected behavior are you seeing?

Our testers are reporting crashes when testing our UrbanAirship-enabled app on a device with its storage full. Not seen this in production; our customers are not having their storages completely full

See below for crash logcat.

### What is the expected behavior?

No crash. Graceful degradation of service if there's no storage.

### What are the steps to reproduce the unexpected behavior?

Launch an app with UrbanAirship integration on a device with full storage.

### Do you have logging for the issue?

```
android.database.sqlite.SQLiteFullException: database or disk is full (code 13)
	at android.database.sqlite.SQLiteConnection.nativeExecute(SQLiteConnection.java)
	at android.database.sqlite.SQLiteConnection.execute(SQLiteConnection.java:553)
	at android.database.sqlite.SQLiteSession.endTransactionUnchecked(SQLiteSession.java:437)
	at android.database.sqlite.SQLiteSession.endTransaction(SQLiteSession.java:401)
	at android.database.sqlite.SQLiteDatabase.endTransaction(SQLiteDatabase.java:522)
	at com.urbanairship.remotedata.RemoteDataStore.savePayloads(RemoteDataStore.java:86)
	at com.urbanairship.remotedata.RemoteData.overwriteCachedData(RemoteData.java:346)
	at com.urbanairship.remotedata.RemoteData.access$100(RemoteData.java:44)
	at com.urbanairship.remotedata.RemoteData$5.run(RemoteData.java:284)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:148)
	at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by rlepinski** [comment]
Thanks for the report. We will take a look and see if we can handle this case better throughout the SDK.

**Comment by rlepinski** [comment]
Fixed in 10.0.0

**Comment by laalto** [comment]
Thanks, will have our testers have another run at this once 10.0.0 is available in the maven repo and can migrate the app to it.
